Test failing for readWriterOptions

Running the test suite on master I get the following error:

```
...
============================================= FAILURES ==============================================
_________________________________________ readWriterOptions _________________________________________
data: /Users/jhereth/repos/github/pyspark-stubs/test-data/unit/sql-session.test:81:
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/pluggy/hooks.py:286: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/pluggy/manager.py:93: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/pluggy/manager.py:87: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/_pytest/runner.py:143: in pytest_runtest_call
    raise e
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/_pytest/runner.py:135: in pytest_runtest_call
    item.runtest()
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/mypy/test/data.py:221: in runtest
    suite.run_case(self)
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/mypy/test/testcheck.py:128: in run_case
    self.run_case_once(testcase)
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/mypy/test/testcheck.py:223: in run_case_once
    assert_string_arrays_equal(output, a, msg.format(testcase.file, testcase.line))
/Users/jhereth/repos/github/pyspark-stubs/.venv/lib/python3.7/site-packages/mypy/test/helpers.py:116: in assert_string_arrays_equal
    raise AssertionError(msg)
E   AssertionError: Unexpected type checker output (/Users/jhereth/repos/github/pyspark-stubs/test-data/unit/sql-session.test, line 81)
--------------------------------------- Captured stderr call ----------------------------------------
Expected:
  main:14: error: Argument "foo" to "load" of "DataFrameReader" has incompatible type "List[str]"; expected "Union[bool, float, int, str]" (diff)
  main:15: error: Argument 2 to "option" of "DataFrameReader" has incompatible type "Tuple[int]"; expected "Union[bool, float, int, str]" (diff)
  main:16: error: Argument "bar" to "options" of "DataFrameReader" has incompatible type "Set[int]"; expected "Union[bool, float, int, str]" (diff)
Actual:
  main:14: error: Argument "foo" to "load" of "DataFrameReader" has incompatible type "List[str]"; expected "Union[bool, float, int, str, None]" (diff)
  main:15: error: Argument 2 to "option" of "DataFrameReader" has incompatible type "Tuple[int]"; expected "Union[bool, float, int, str, None]" (diff)
  main:16: error: Argument "bar" to "options" of "DataFrameReader" has incompatible type "Set[int]"; expected "Union[bool, float, int, str, None]" (diff)

Alignment of first line difference:
  E: ...on[bool, float, int, str]"
  A: ...on[bool, float, int, str, None]"
                                ^
```

--- a/test-data/unit/sql-session.test
+++ b/test-data/unit/sql-session.test
@@ -83,16 +83,16 @@ from pyspark.sql import SparkSession
 
 spark = SparkSession.builder.getOrCreate()
 
-spark.read.option("foo", True).option("foo", 1).option("foo", 1.0).option("foo", "1")
-spark.readStream.option("foo", True).option("foo", 1).option("foo", 1.0).option("foo", "1")
+spark.read.option("foo", True).option("foo", 1).option("foo", 1.0).option("foo", "1").option("foo", None)
+spark.readStream.option("foo", True).option("foo", 1).option("foo", 1.0).option("foo", "1").option("foo", None)
 
-spark.read.options(foo=True, bar=1).options(foo=1.0, bar="1")
-spark.readStream.options(foo=True, bar=1).options(foo=1.0, bar="1")
+spark.read.options(foo=True, bar=1).options(foo=1.0, bar="1", baz=None)
+spark.readStream.options(foo=True, bar=1).options(foo=1.0, bar="1", baz=None)
 
 spark.read.load(foo=True)
 spark.readStream.load(foo=True)
 
-spark.read.load(foo=["a"])  # E: Argument "foo" to "load" of "DataFrameReader" has incompatible type "List[str]"; expected "Union[bool, float, int, str]"
-spark.read.option("foo", (1, ))  # E: Argument 2 to "option" of "DataFrameReader" has incompatible type "Tuple[int]"; expected "Union[bool, float, int, str]"
-spark.read.options(bar={1})  # E: Argument "bar" to "options" of "DataFrameReader" has incompatible type "Set[int]"; expected "Union[bool, float, int, str]"
+spark.read.load(foo=["a"])  # E: Argument "foo" to "load" of "DataFrameReader" has incompatible type "List[str]"; expected "Union[bool, float, int, str, None]"
+spark.read.option("foo", (1, ))  # E: Argument 2 to "option" of "DataFrameReader" has incompatible type "Tuple[int]"; expected "Union[bool, float, int, str, None]"
+spark.read.options(bar={1})  # E: Argument "bar" to "options" of "DataFrameReader" has incompatible type "Set[int]"; expected "Union[bool, float, int, str, None]"
 [out]

